### DISCUSSION TITLE
allow passing url of croissant json to mlc.Dataset(file=url)

### DISCUSSION THREAD

**Discussion by luisoala** [mlcommons/croissant#407] [Ideas]
i.e. right here https://github.com/mlcommons/croissant/blob/77fd5a7183f89b7f292fd90a5a04cf41e60baf03/python/mlcroissant/mlcroissant/_src/datasets.py#L48

for user ergonomics, would it not make more sense to just pass url to the json (e.g. https://github.com/mlcommons/croissant/blob/c3168c5d2b0203b92c2419128176bfb11c163370/datasets/coco2014/metadata.json) rather than having user first download the json and the pass the local file path

**Comment by marcenacp** [comment]
Hey @luisoala, the feature is now implemented! Closing this discussion :)

**Reply by luisoala** [reply]
🥳🥳
